File: backend/app/Models/Tactic.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Tactic extends Model
{
    use HasFactory;

    protected $fillable = ['name', 'formation', 'team_id', 'is_default'];

    public function team()
    {
        return $this->belongsTo(Team::class);
    }

    public function playerPositions()
    {
        return $this->hasMany(PlayerPosition::class);
    }

    public function slotPositions()
    {
        return $this->hasMany(TacticSlotPosition::class);
    }

    public function tacticalInstructions()
    {
        return $this->hasMany(TacticalInstruction::class);
    }
}

File: frontend/lib/models/tactic.dart
class Tactic {
  final int id;
  final String name;
  final String formation;
  final int? teamId;
  final bool isDefault;

  Tactic({
    required this.id,
    required this.name,
    required this.formation,
    this.teamId,
    required this.isDefault,
  });

  factory Tactic.fromJson(Map<String, dynamic> json) {
    return Tactic(
      id: json['id'] as int? ?? 0,
      name: json['name'] as String? ?? 'Unknown',
      formation: json['formation'] as String? ?? 'Unknown',
      teamId: json['team_id'] as int?,
      isDefault: json['is_default'] == true || json['is_default'] == 1,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'formation': formation,
      'team_id': teamId,
      'is_default': isDefault,
    };
  }
}
